connector/spanmetricsconnector/connector.go

```diff
@@ -255,8 +259,10 @@ func (p *connectorImp) resetState() {
 		p.metricKeyToDimensions.RemoveEvictedItems()
 
 		// Exemplars are only relevant to this batch of traces, so must be cleared within the lock
-		for _, m := range p.resourceMetrics {
-			m.histograms.Reset(true)
+		if !p.config.Histogram.Disable {
```

Comment:
I think I did not get the point...
If we make for example variable: `p.config.Histogram.Enable`, there is not way to make it true by default, or it may cause huge refactoring in tests an possbile. 
Early return in most places could not be used because function works with both type of metrics histogram and sum, but we disable only histogram.

Reply:
Ah, I meant:

```go
if p.config.Histogram.Disable {
    return
}

for _, m := ...
```

Reply:
Unfortunately we can not do early return in these cases, because we  check metrics in loops, and by making early return in these places we will affect sums metrics as well.

Of cause we can use somehting like this in loops:
```
if p.config.Histogram.Disable {
 continue
}
```

but i'm not sure that it makes code more readable

Reply:
Actually only in this case we can use early return. 

Fixed. 

For other cases please see my comment above.